sparkles/core.py

```diff
@@ -1251,3 +1254,17 @@ def check_fid_count(self):
         if self.n_fid != typical_n_fid:
             msg = f'{obs_type} requested {self.n_fid} fids but {typical_n_fid} is typical'
             self.add_message('caution', msg)
+
+    @classmethod
+    def from_ocat(cls, obsid, t_ccd=-5, man_angle=5, date=None, roll=None):
+        if date is None:
+            date = get_ocat_obsid_date(obsid)
+        if date is None:
+            date = CxoTime().now()
```

Comment:
Why are you doing the separate OCAT query to get the date when `get_params_yoshi` will do the same again? This is hitting the server twice by default.

Also it looks like the second `if date is None` is leftover.

Reply:
This needs a comment because the second "if date is None" was intended for "user did not supply date *and* the observation has not been assigned an lts bin" (not uncommon).

Reply:
For example:
```
acar = ACAReviewTable.from_ocat(...)
acar.run_aca_review(make_html=True, roll_level='critical')
```